src/utilities/report-utils.ts

```diff
@@ -67,34 +67,44 @@ export const getReportUrl = (questionRefId?: string) => {
     result += "&runKey=" + runKey
             + "&activity=" + resourceUrl
             + "&resourceUrl=" + resourceUrl;
-  }
-  else {
+    return result;
+  } else {
     // We know this is a IPortalData because there is no runKey
     const portalData = getPortalData() as IPortalData;
-    const classInfoUrl = portalData?.portalJWT?.class_info_url;
-    const authDomainUrl = classInfoUrl?.split("/api")[0];
-    const offeringBaseUrl = classInfoUrl?.split("/classes")[0]+"/offerings/";
-    const offeringId = portalData?.offering.id;
-    const offeringUrl = encodeURIComponent(offeringBaseUrl + offeringId);
-    const studentId = portalData?.platformUserId;
+    // Handles logged in teachers that are previewing a non-assigned resource or are in teacher-edition mode
+    if (!portalData?.offering.id) {
+      return null;
+    } else {
+      const classInfoUrl = portalData?.portalJWT?.class_info_url;
+      const authDomainUrl = classInfoUrl?.split("/api")[0];
+      const offeringBaseUrl = classInfoUrl?.split("/classes")[0]+"/offerings/";
+      const offeringId = portalData?.offering.id;
+      const offeringUrl = encodeURIComponent(offeringBaseUrl + offeringId);
+      const studentId = portalData?.platformUserId;
 
-    // In the future we should be able to drop this because the portal
-    // report should be able to get all the info it needs from the
-    // offering url
-    result += "&class=" + encodeURIComponent(classInfoUrl || "")
-            + "&offering=" + offeringUrl
-            + "&reportType=offering&studentId="+studentId
-            + "&auth-domain="+authDomainUrl;
-  }
+      // In the future we should be able to drop this because the portal
+      // report should be able to get all the info it needs from the
+      // offering url
+      result += "&class=" + encodeURIComponent(classInfoUrl || "")
+              + "&offering=" + offeringUrl
+              + "&reportType=offering&studentId="+studentId
+              + "&auth-domain="+authDomainUrl;
 
-  if (questionRefId) {
-    // Limit report to a single question.
-    result += "&iframeQuestionId=" + refIdToAnswersQuestionId(questionRefId);
+      if (questionRefId) {
+        // Limit report to a single question.
+        result += "&iframeQuestionId=" + refIdToAnswersQuestionId(questionRefId);
+      }
+      return result;
+    }
   }
-
-  return result;
 };
 
+export const isValidReportLink = getReportUrl() != null;
```

Comment:
I think this should probably be a function instead of a static property. `getReportUrl` is based off of the portalData. So by calling getReportUrl statically it means this module is probably going to trigger the first call to getPortalData. That might happen too soon if the logic for how getPortalData is implemented is changed. 

So it seems safer to make this a function and then have it just call getReportUrl every time this function is called. If that is a problem it could be optimized to cache the result since the status of `isValidReportLink` shouldn't change for the same params and portal data.